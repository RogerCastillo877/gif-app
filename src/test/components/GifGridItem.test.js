import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('should prube <GifGridItem />', () => {

    const title = 'titulo';
    const url = 'https://miss.com';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } />)
    
    test('should show the component correct', () => {
        
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a paraghraf with title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    });

    test('should have a img same to url and alt of props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title );
    });

    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes('animate__fadeIn')).toBe(true);
    });
});