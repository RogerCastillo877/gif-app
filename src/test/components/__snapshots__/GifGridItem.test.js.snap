// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should prube <GifGridItem /> should show the component correct 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://miss.com"
  />
  <p>
    titulo
  </p>
</div>
`;
